src/main/java/com/idrsolutions/microservice/BaseServlet.java

```diff
@@ -267,19 +269,22 @@ protected void doPost(final HttpServletRequest request, final HttpServletRespons
     }
 
     /**
-     * Sanitize the file name by removing all none url/filepath friendly
-     * characters.
+     * Sanitize the file name by removing all non filepath friendly characters.
+     *
+     * Allow only characters valid across all (most) platforms.
+     * Note that this does not cover all reserved filenames on Windows (E.g. CON, COM1, LTP1, etc), therefore it
+     * remains possible for a user to pass a file that cannot be stored if the server is running on Windows.
+     *
+     * The space character is also currently replaced with an underscore because file names that consist only of
+     * spaces and file paths that end with spaces are not allowed on Windows.
+     *
+     * More info: https://stackoverflow.com/a/31976060
      *
      * @param fileName the filename to sanitize
      * @return the sanitized filename
      */
     private static String sanitizeFileName(final String fileName) {
-        final int extPos = fileName.lastIndexOf('.');
-        // Limit filenames to chars allowed in unencoded URLs and Windows filenames for now
-        final String fileNameWithoutExt = fileName.substring(0, extPos).replaceAll("[^$\\-_.+!'(),a-zA-Z0-9]", "_");
-        final String ext = fileName.substring(extPos + 1);
-
-        return fileNameWithoutExt + '.' + ext;
+        return fileName.replaceAll("[\\\\/:\"*?<>| \\p{Cc}]", "_");
```

Comment:
Can I just check that quotation mark, it has been escaped for the java string, however I believe as a delimiter it needs to be escaped inside the regex string, should it look more like this:
```suggestion
        return fileName.replaceAll("[\\\\/:\\\"*?<>| \\p{Cc}]", "_");
```

Reply:
Nope, it works correctly as-is and IntelliJ complains about redundant escaping if I double escape it.
```java
System.out.println("\"".replaceAll("[\\\\/:\"*?<>| \\p{Cc}]", "_")); // Prints _
```

Reply:
So it does, that's a failing from Regex101 then
![image](https://user-images.githubusercontent.com/87422941/142470280-baa55cd7-2dff-4fd3-aeb0-ffc8adc6fc14.png)